spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubsubAutoConfiguration.java

```diff
@@ -112,4 +117,42 @@ public PublisherFactory defaultPublisherFactory(GcpProjectIdProvider projectIdPr
 		return new DefaultPublisherFactory(projectIdProvider, subscriberProvider, channelProvider,
 				credentialsProvider);
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public PubsubAdmin pubsubAdmin(GcpProjectIdProvider projectIdProvider,
+			TopicAdminClient topicAdminClient,
+			SubscriptionAdminClient subscriptionAdminClient) {
+		return new PubsubAdmin(projectIdProvider, topicAdminClient, subscriptionAdminClient);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public TopicAdminClient topicAdminClient(CredentialsProvider credentialsProvider) {
+		try {
+			return TopicAdminClient.create(
+					TopicAdminSettings.defaultBuilder()
+							.setCredentialsProvider(credentialsProvider)
+							.build());
+		}
+		catch (IOException ioe) {
+			throw new PubsubException("An error occurred while creating TopicAdminClient.", ioe);
+		}
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public SubscriptionAdminClient subscriptionAdminClient(
+			CredentialsProvider credentialsProvider) {
+		try {
+			return SubscriptionAdminClient.create(
+					SubscriptionAdminSettings.defaultBuilder()
+							.setCredentialsProvider(credentialsProvider)
```

Comment:
Can't we have an additional `PubsubAdmin` ctor based on the `CredentialsProvider` to instantiate `TopicAdminClient` and `SubscriptionAdminClient` internally similar way?

Just in case for flexibility to let end-user to avoid this boilerplate code